pyartcd/pyartcd/locks.py

```diff
@@ -1,54 +1,69 @@
+import enum
 import logging
 
 from aioredlock import Aioredlock
 
 from pyartcd import redis
 
+
+# Defines the pipeline locks managed by Redis
+class Lock(enum.Enum):
+    def __str__(self):
+        return f'{self.value}-lock'
+
+    OLM_BUNDLE = 'olm-bundle'
+    MIRRORING_RPMS = 'mirroring-rpms'
+    COMPOSE = 'compose'
+    GITHUB_ACTIVITY = 'github-activity'
+    MASS_REBUILD = 'mass-rebuild-serializer'
+    SIGNING = 'signing'
+
+
 # This constant defines for each lock type:
 # - how many times the lock manager should try to acquire the lock before giving up
 # - the sleep interval between two consecutive retries, in seconds
 # - a timeout, after which the lock will expire and clear itself
 LOCK_POLICY = {
-    # default policy: give up after 1 hour
-    'default': {
-        'retry_count': 36000,
-        'retry_delay_min': 0.1,
-        'lock_timeout': 60 * 60 * 1,  # 1 hours
-    },
     # olm-bundle: give up after 1 hour
-    'olm_bundle': {
+    Lock.OLM_BUNDLE: {
         'retry_count': 36000,
         'retry_delay_min': 0.1,
         'lock_timeout': 60 * 60 * 2,  # 2 hours
     },
     # mirror RPMs: give up after 1 hour
-    'mirroring_rpms': {
+    Lock.MIRRORING_RPMS: {
         'retry_count': 36000,
         'retry_delay_min': 0.1,
         'lock_timeout': 60 * 60 * 3,  # 3 hours
     },
     # compose: give up after 1 hour
-    'compose': {
+    Lock.COMPOSE: {
         'retry_count': 36000,
         'retry_delay_min': 0.1,
         'lock_timeout': 60 * 60 * 6,  # 6 hours
     },
-    # ocp4: give up after 1 hour
-    'ocp4': {
+    # github-activity-lock: give up after 1 hour
+    Lock.GITHUB_ACTIVITY: {
         'retry_count': 36000 * 1,
         'retry_delay_min': 0.1,
-        'lock_timeout': 60 * 60 * 6,  # 12 hours
+        'lock_timeout': 60 * 60 * 6,  # 6 hours
     },
     # mass rebuild: give up after 8 hours
-    'mass_rebuild': {
+    Lock.MASS_REBUILD: {
         'retry_count': 36000 * 8,
         'retry_delay_min': 0.1,
         'lock_timeout': 60 * 60 * 12,  # 12 hours
+    },
+    # signing: give up after 1 hour
+    Lock.SIGNING: {
+        'retry_count': 36000,
+        'retry_delay_min': 0.1,
+        'lock_timeout': 60 * 60 * 1,  # 1 hour
     }
 }
 
 
-class LockManager(Aioredlock):
+class _LockManager(Aioredlock):
```

Comment:
I think it is ok to keep this type public since the type itself should be visible to other modules.
I guess your purpose is to discourage instantiating it directly. Maybe new_lock_manager can be moved into the class as a static method? Something like:
```python
class LockManager:
  @staticmethod
  def from_lock(policy: Lock):
    return ...
```

Reply:
Done with https://github.com/openshift-eng/aos-cd-jobs/pull/3874/commits/0cef36c15b13544714f48a35426110e9b583fafe